automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/apps/AppCreationSpec.scala

```diff
@@ -25,92 +31,112 @@ class AppCreationSpec extends GPAllocFixtureSpec with LeonardoTestUtils with GPA
-        customEnvironmentVariables = Map("WORKSPACE_NAME" -> "Galaxy-Workshop-ASHG_2020_GWAS_Demo")
+  val createGalaxyAppRequest = defaultCreateAppRequest.copy(
+    diskConfig = Some(
+      PersistentDiskRequest(
+        Generators.genDiskName.sample.get,
+        Some(DiskSize(300)),
+        None,
+        Map.empty
       )
-
-      LeonardoApiClient.client.use { implicit client =>
-        for {
-          _ <- loggerIO.info(s"AppCreationSpec: About to create app ${googleProject.value}/${appName.value}")
-
-          rat <- Ron.authToken()
-          implicit0(auth: Authorization) = Authorization(Credentials.Token(AuthScheme.Bearer, rat.value))
-
-          // Create the app
-          _ <- LeonardoApiClient.createApp(googleProject, appName, createAppRequest)
-
-          // Verify the initial getApp call
-          getApp = LeonardoApiClient.getApp(googleProject, appName)
-          getAppResponse <- getApp
-          _ = getAppResponse.status should (be(AppStatus.Provisioning) or be(AppStatus.Precreating))
-
-          // Verify the app eventually becomes Running
-          _ <- IO.sleep(120 seconds)
-          monitorCreateResult <- streamUntilDoneOrTimeout(
-            getApp,
-            120,
-            10 seconds,
-            s"AppCreationSpec: app ${googleProject.value}/${appName.value} did not finish creating after 20 minutes"
-          )(implicitly, appInStateOrError(AppStatus.Running))
-          _ <- loggerIO.info(
-            s"AppCreationSpec: app ${googleProject.value}/${appName.value} monitor result: ${monitorCreateResult}"
-          )
-          _ = monitorCreateResult.status shouldBe AppStatus.Running
-
-          _ <- IO.sleep(1 minute)
-
-          // Delete the app
-          _ <- LeonardoApiClient.deleteApp(googleProject, appName, false)
-
-          // Verify getApp again
-          getAppResponse <- getApp
-          _ = getAppResponse.status should (be(AppStatus.Deleting) or be(AppStatus.Predeleting))
-
-          // Verify the app eventually becomes Deleted
-          // Don't fail the test if the deletion times out because the Galaxy pre-delete job can sporadically fail.
-          // See https://broadworkbench.atlassian.net/browse/IA-2471
-          listApps = LeonardoApiClient.listApps(googleProject, true)
-          implicit0(deletedDoneCheckable: DoneCheckable[List[ListAppResponse]]) = appDeleted(appName)
-          monitorDeleteResult <- streamFUntilDone(
-            listApps,
-            120,
-            10 seconds
-          ).compile.lastOrError
-          // TODO remove first case in below if statement when Galaxy deletion is reliable
-          _ <- if (!deletedDoneCheckable.isDone(monitorDeleteResult)) {
-            loggerIO.warn(
-              s"AppCreationSpec: app ${googleProject.value}/${appName.value} did not finish deleting after 20 minutes. Result: $monitorDeleteResult"
+    ),
+    customEnvironmentVariables = Map("WORKSPACE_NAME" -> "Galaxy-Workshop-ASHG_2020_GWAS_Demo")
```

Comment:
Should the appType of Galaxy be specified? Perhaps that is the default, but it would be more explicit if it was specified (and I would hope we keep moving away from the Galaxy default assumption).

Reply:
Nit: I would also suggest creating a helper method for createAppRequest since it looks like the only difference between the Galaxy one and the Cromwell one is the appType and the workspace name (which could be provided as input arguments).